### PR TITLE
Resolution Audit S6C: scalar history cleanup

### PR DESCRIPTION
**Pull request by canfieldjuan**
Plan: plans/PR-Resolution-Audit-S6C-Scalar-History.md
Slice phase: Production hardening

Scalar ticket-history exports took the ordinary one-message compaction path, erasing transcript boundaries and admitting signatures or quoted replies as customer evidence. This fixes the root at the scalar-history admission choke point with one evidence-derived event model and state policy shared by emitted text and junk admission. Ambiguous sender-shaped prose stays customer content; corroborated HTML and original-message reply evidence now survives normalization; quote and signature runs resume at the accepted blank or structural customer boundaries. Closes #2044.

## Intentional
- Only scalar `ticket_history`, `history`, and `conversation_history` aliases use transcript cleanup; ordinary bodies, scalar one-message comments, and structured comment lists retain their paths.
- Customer-content preservation wins ambiguous cases. An uncorroborated Unicode or ASCII name in `On ... wrote:` remains content; email, original-message metadata, or quote markers are affirmative reply evidence.
- Quote markers are stripped only for detection. Standalone customer `>` lines retain their original text.
- Signature delimiters require a Unicode title-cased head plus bounded contact evidence. Mobile signatures remain exact markers.
- Bare blank resumption is retained because accepted issue #2044 explicitly defines blank as an end to quote/signature exclusion; the contrary review request is waived rather than changing the accepted contract.
- No privacy, evidence-tier, status/outcome, downstream scrubber, or customer-facing product-shape semantics change.

## Deferred
- #2050 owns bounded status/outcome synonym normalization.
- #2045 owns customer evidence-tier and customer-wording semantics after S6C.
- #2037 remains paused evidence and is not modified or revived.

## Parked hardening
- None.

## Cold diff reconstruction
- Changed: `extracted_content_pipeline/support_ticket_input_package.py:120-150` defines bounded positive-evidence grammars and the shared event policy without sender vocabulary.
- Changed: `extracted_content_pipeline/support_ticket_input_package.py:708-728` preserves raw scalar-history presence while feeding only sanitized history text to the existing junk gate, retaining auto-reply/bounce precedence.
- Changed: `extracted_content_pipeline/support_ticket_input_package.py:869-914` routes only the three scalar-history aliases through the sanitizer in both raw-comment and emitted-text assembly; ordinary and structured comments retain their paths.
- Changed: `extracted_content_pipeline/support_ticket_input_package.py:917-947` protects angle-bracket email text before HTML extraction, preserves blank events, derives one event per line, and applies the shared state policy.
- Changed: `extracted_content_pipeline/support_ticket_input_package.py:978-1011` scans a bounded run of arbitrary RFC-shaped header fields for `From:` plus `Sent:`/`Date:` and retains evidence-gated `On ... wrote:` detection.
- Changed: `tests/test_extracted_support_ticket_input_package.py:1925-2006` generates 577 alias/separator/evidence/marker/header/state-boundary cases through `build_support_ticket_input_package`, including HTML angle emails, long header runs, Unicode, nested quote markers, HTML/Unicode blanks, and both error directions.
- Changed: `tests/test_extracted_support_ticket_input_package.py:2009-2043` proves all-quote junk admission and unchanged non-history scalar-comment behavior through the real package entrypoint.
- Changed: `plans/PR-Resolution-Audit-S6C-Scalar-History.md:1` records the resolved contract, approved fix-two/waive-one disposition, verification, and 399-LOC budget.
- Contract match: raw evidence preservation, bounded header scanning, structural state transitions, later-customer resumption, and default-to-content ambiguity handling map one-for-one to accepted #2044.
- Gaps: None. No contract item is missing, no change is untraced, and no forbidden adjacent lane or product surface changed.

## AI reconciliation
- [chatgpt-codex-connector] Preserve raw sender emails before HTML line extraction: fixed by entity-protecting angle-bracket emails before line-preserving HTML extraction; plain, `<br>`, and `<br/>` separators are generated across all aliases and evidence combinations.
- [chatgpt-codex-connector] Scan complete original-message header blocks: fixed with a bounded 12-line run of arbitrary RFC-shaped header names; zero and five intermediate headers plus `Sent:`/`Date:` are generated with quote-marker variants across every alias.
- [chatgpt-codex-connector] Do not resume skipped runs on a bare blank: waived with operator approval because accepted issue #2044 requires quote/signature exclusion only until blank or a likely new-message boundary. The oracle continues to prove plain, Unicode, and HTML blank resumption in both modes.
- Prior sender-vocabulary findings remain superseded by the resolved evidence-required contract; no guessed person identity controls reply admission.

All findings fixed or waived: yes.

## Verification
- `python -m pytest -q tests/test_extracted_support_ticket_input_package.py -k 's6c_scalar_history or s6c_non_history'` - 3 passed, 243 deselected; 577 generated event/state cases.
- `python -m pytest -q tests/test_extracted_support_ticket_input_package.py` - 246 passed.
- Exact extracted-content maturity ratchet - passed; changed file stayed at its pre-PR score of 9.
- `bash scripts/validate_extracted_content_pipeline.sh` - passed.
- `python extracted/_shared/scripts/forbid_atlas_reasoning_imports.py extracted_content_pipeline` - clean.
- `python scripts/audit_extracted_standalone.py --fail-on-debt` - zero findings.
- `bash scripts/check_ascii_python.sh` - passed.
- `bash scripts/run_extracted_pipeline_checks.sh` - 10,732 passed, 21 skipped; one pre-existing `pynvml` warning.

## Diff size
3 files, +396 / -3 (399 changed LOC)

### CODE REVIEW

**Review by canfieldjuan — COMMENTED** [review]
**Review at `04a3de86` (current head; the surfaced `8ff3e15f` was superseded, all 22 bot threads now resolved). This is a genuine net improvement and safe to merge -- LGTM (pending `unit-gate`); `claude-review=success`. But it is a #2061-class open-input classifier and I want the convergence approach on record. Pure-reviewer.**

**Why it is safe to merge (the load-bearing reconstruction):** `_scalar_history_text` is fail-open -- it admits every line by default and only drops runs it recognizes as quoted-reply / original-message / signature / footer. I traced the scalar path end to end: `_public_ticket_row` does NOT scrub (it only strips `_`-prefixed keys), and `support_ticket_comment_is_private` guards the STRUCTURED list-comment path, not the scalar path -- so this heuristic IS the scalar-history boundary. Normally a fail-open boundary on PII-bearing text would be a blocker, but here it is strictly better than `main` by construction: today `_comments_text` compacts the whole scalar-history blob as one admitted comment (every signature/quote/footer already reaches the report), so this scanner can only admit a SUBSET. It cannot introduce a leak; it reduces the existing one. Structured-comment privacy and row-level `support_ticket_row_is_private` are unchanged. 17 S6C/scalar-history tests pass at head (my run); live-reconciliation and the deflection maturity sweep are green.

**Structural note (the durable point, not a merge blocker):** this scanner consumed ~22 Codex boundary findings (User-Agent-as-customer, dash-as-signature, HTML-entity blanks, post-footer resumes, first-person footer resumes, quote-entry sender evidence, ...), and the committed tests are fixture-parametrized (`test_s6c_scalar_history_*`), not a grammar-derived property test with an independent oracle. That is the exact enumerate-and-miss shape `docs/GUARD_CLASS_CLOSURE.md` was written for (from the #2061 arc): a fixture-driven open-input classifier does not close the CLASS -- the next unseen signature/quote/footer shape will still leak (under-exclusion) or drop a real follow-up (over-exclusion). Because this slice is a quality improvement over admit-everything and the repo's `guard-class-closure-lint` is green, I am not blocking on it -- but the durable close is a generated cross-product / property test over the signature / quote / footer / customer-cue grammars with a semantic oracle, as a fast-follow. Landing that would convert the 22-round patch history into a by-construction guarantee and stop the next boundary shape from reopening this.

**Residual risks (both directions, bounded):** under-exclusion (unrecognized signature/quote with PII admitted -- still <= main); over-exclusion (an unseen shape wrongly ends a customer follow-up, losing FAQ/resolution signal -- guarded partially by the keep-followup fixtures). Neither exceeds the status quo.

boundary-probe: exclusion swept both sides, but via fixtures not a grammar oracle, so the class is not closed -- residual unseen shapes remain on both sides, bounded by the subset-of-main invariant. effect-trace: the scanner emits a subset of main's admitted scalar text into `source_material` with no downstream scrub, so it strictly reduces (never increases) signature/quote leakage; structured-comment privacy path is untouched. State: MERGEABLE; live-reconciliation / diff-budget / maturity sweeps green; `unit-gate` pending. Verdict: LGTM -- owner merges on unit-gate green; recommend the property-test fast-follow.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `0904c7a8` -- LGTM holds. This round is more heuristic refinement (split the reply-header regex into date/email/time-sender parts, added name/contact/needing-phrase recognizers), and the safety invariant is preserved. `claude-review=success`. Pure-reviewer.**

Verified the refactor did not change the load-bearing property: the scanner still admits by default (`admitted.append(line)` is the loop's fall-through) and only drops runs it affirmatively recognizes as quote/signature/footer. So it remains a strict SUBSET of `main`'s admit-everything scalar-history behavior -- no new leak is possible regardless of the pattern edits; only the exclusion precision moved. 17 S6C/scalar-history tests pass at head (my run); 0 unresolved threads; live-reconciliation / diff-budget / deflection maturity sweep green; unit-gate pending.

**Structural note stands (now more strongly):** this is another pattern-tuning round on a fail-open open-input classifier whose tests are still fixture-parametrized, not a grammar-derived property test with an independent oracle. The refactor makes the recognizers cleaner but does not close the class -- the next unseen signature/quote/footer shape can still under-exclude (<= main) or over-exclude a real follow-up. Per `docs/GUARD_CLASS_CLOSURE.md`, the durable end to this refinement sequence is a generated cross-product / oracle test over the signature/quote/footer/customer-cue grammars; landing that would let you stop tuning regexes round over round. Not blocking (net improvement + guard-class-closure-lint green), but it is the highest-leverage next step. Verdict: LGTM -- owner merges on unit-gate green.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `7feeac8a` -- safety invariant still holds, but this round the complexity note stopped being hypothetical: the `extracted-content-pipeline` maturity ratchet went RED. LGTM on safety (advisory ratchet does not block); `claude-review=success`. Pure-reviewer.**

Safety unchanged and verified: `_scalar_history_text` still admits by default (`admitted.append(line)` fall-through) and only drops recognized runs, so it stays a strict subset of `main`'s admit-everything -- no new leak. The `had_source_text` fix to count scalar-only rows is a correct over-exclusion reduction. 257 support-ticket tests pass at head (my run); 0 unresolved threads; live-reconciliation and the deflection-content-ops ratchet green.

**New signal (advisory, non-blocking, but it is the point):** `Maturity sweep ratchet (extracted-content-pipeline)` FAILED this round -- it passed at `04a3de86`. Adding `_HISTORY_SIGNATURE_COMPANY_RE`, `_HISTORY_VALEDICTION_RE`, the context-aware reply-header lookahead, etc. pushed this module's structural brittleness across its committed baseline. This is the concrete cost of converging the classifier by accreting per-shape regexes: each boundary round adds branches, and the brittleness ratchet has now caught up with it. It is advisory so it does not gate the merge, but it should not be left red, and the fix is the same one I have flagged: a grammar-derived property/oracle test lets the scanner be expressed as one grammar + one oracle instead of N special-case recognizers, which both closes the class AND gets the ratchet back under baseline. That is now the highest-leverage move on this PR by two independent measures (class-closure and the brittleness gate).

Verdict: LGTM on safety/behavior; owner merges on unit-gate green, but please either land the oracle refactor or justify/greenlight the extracted-content-pipeline ratchet before merge so it is not shipped red.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `8e44fc87` (round 5). The ratchet fix worked and unit-gate is green -- but round 5 fixing the ratchet immediately produced 3 more Codex boundary findings, and that is the whole point. Not blocking on safety, but I am not greening past open real findings: `claude-review=failure` until the loop is closed the right way. Pure-reviewer.**

**Good this round:** the one-line valediction-start simplification got `Maturity sweep ratchet (extracted-content-pipeline)` back to GREEN, and `unit-gate` now passes (7m51s). Safety invariant unchanged: still admit-by-default, still a strict subset of `main`'s admit-everything, so no new leak.

**The 3 new Codex P2s are real, and all the same class** (`support_ticket_input_package.py:980/990/1009`): a display-name reply header `On Monday, Jane wrote:` without an email/time suffix, a blank line before signature-tail evidence, and a signature with a job-title line between name and company. Each is an unrecognized shape that the fail-open scanner therefore admits -- i.e. under-exclusion, bounded by <= main (no new leak vs today), so they are quality-completeness, not safety regressions. I confirmed they are genuine, not noise.

**But this is now round 5 with ~25 cumulative boundary findings, and round 5 demonstrably generated 3 new ones the moment it closed the ratchet.** That is proof the enumerate-and-patch approach is non-convergent: Codex will keep producing the next unlisted signature/quote shape indefinitely, and `live-reconciliation` will keep going red each round. Per the review-loop cap discipline, the choice is now binary and neither branch is another blind patch round:

1. **Close the class by construction** -- land the grammar-derived property/oracle test I have flagged since round 1. Define the recognized reply-header / signature / footer grammar explicitly and test the whole cross-product against a semantic oracle. Codex's individual examples then either fall inside the defined grammar (handled) or are explicitly out-of-grammar (documented), and the finding stream terminates. This is the only convergent path.
2. **Accept the bounded heuristic as-is** -- treat the current scanner as a best-effort quality improvement (safe: <= main), document the residual under-exclusion class as a known limitation in the plan, reconcile the open threads as acknowledged-not-fixing-this-round, and merge on green. This is legitimate ONLY because the safety net is subset-of-main, not this heuristic.

Continuing to add one regex per Codex example is option 3, and it will not terminate. I would take option 1. Verdict: needs the loop closed (open real findings + reconciliation red); the code is safe, but do not ship it by out-patching Codex one shape at a time.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `07115598` (round 6). This is the right move -- you took option 1 and built a genuine grammar + independent semantic oracle, and the ratchet is green again. But it surfaced the direction I had under-weighted: the grammar now OVER-excludes real customer content, and three of the new findings are true regressions, not <=main quality noise. MAJOR; `claude-review=failure`. Pure-reviewer.**

**Credit where due (the structural win):** `_scalar_history_text` is now a bounded reply-header/signature grammar with one skip-policy decision instead of per-example branches, and `test_s6c_scalar_history_generated_grammar_matches_semantic_oracle` is a real generated cross-product whose verdicts come from an INDEPENDENT semantic role oracle (`sender_oracle`/`terminal_oracle` assign keep/drop by meaning, then `assert ... is (role == 'content')`) -- not parity with the implementation. `Maturity sweep ratchet (extracted-content-pipeline)` is green, the oracle test passes, and the admit-by-default subset-of-main invariant still holds for the leak direction. This is the convergent shape I asked for.

**Why it still blocks -- the over-exclusion side the subset-of-main net does NOT cover.** subset-of-main bounds under-exclusion (leaks <= today); it does nothing for over-exclusion, where the grammar drops content `main` keeps. Three new Codex P2s are confirmed content-loss regressions:
- `:131` -- `On Monday, I wrote:\nError 500. How do I fix it?`: the display-name sender production accepts the bare pronoun `I`, so ordinary first-person prose enters terminal quote mode and the customer's actual question is dropped. For a FAQ-mining product, losing the question is the worst possible miss.
- `:143` -- `I need help with the export.` after a real signature is still skipped because the post-signature `need` resume only allows `need` + article/pronoun, not `need help`.
- `:1015` -- a customer detail block after `--` that starts with a URL/phone (`https://api.example.com/webhook returns 500.`) is treated as signature evidence and dropped.
These violate the plan's own `Must not change: ... keep followup / resolution evidence`.

**The fix stays on the convergent path -- extend the ORACLE, do not revert to per-example regex.** The oracle is independent but not yet class-complete: its enumeration used real senders (`Jane Agent`, `it`, `the system`) and never a bare pronoun in sender position, and its terminals never a URL/phone content block or a `need help` resume. Add those as oracle productions with their correct semantic verdict (pronoun-in-sender -> content; `need help` -> resume; URL/phone-first block after `--` -> content), let the failing generated cases drive the grammar correction, and the three regressions close by construction while the class stays covered on both sides. That keeps you out of the round-by-round patch loop.

boundary-probe: both sides now exercised by the generated oracle, but the oracle's enumeration is incomplete on the over-exclusion side (pronoun sender, need-help resume, contact-first content) -- that is exactly where the three regressions live. effect-trace: the grammar's sender/signature productions are over-broad, so real customer questions/follow-ups/details are classified as quote/signature and dropped from `source_material` -> lost FAQ input; this is content loss vs `main`, not bounded by subset-of-main. Verdict: MAJOR -- great structural direction, but fix the over-exclusion false-positives via the oracle before merge; do not ship it dropping customer questions.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `9790860f` (round 7). You fixed my three regressions the right way -- by extending the oracle -- and I verified all three close (`:131` keeps the question, `:143` keeps 'I need help', `:1015` keeps the URL detail). But two new content-loss regressions remain, and the shape of them is the actionable lesson. MAJOR; `claude-review=failure`. Pure-reviewer.**

**Confirmed by direct probe at head (both drop real content vs `main`):**
- `:129` -- `On Monday 09:15 I wrote:\nError 500. How do I fix it?` -> the ROW is dropped entirely. This is the SAME lost-question bug as `:131`, but through the time+name sender production. You corrected the display-name production to reject the bare pronoun `I`, and the oracle now asserts pronoun->content -- but only for that one production. The parallel time+name sender still accepts `I`, so the identical semantic rule fails on a parallel path.
- `:738` -- a row with `resolution_text="We issued the refund on Tuesday."` and a quote-only history -> the ROW is dropped as `no_new_content`. The junk-gate fallback you added THIS round (`junk_reason is None and raw_scalar_history and not gate_body`) fires before explicit evidence like `resolution_text` is considered, so a row carrying real resolution evidence is discarded. This is a regression introduced by this round's change.
- `:1030` -- one-word company line in a signature not skipped: under-exclusion, <= main, bounded/non-blocking.

**The structural fix (so round 8 does not reopen round 7):** the oracle is being extended per-production, reactively -- pronoun-is-content was proven for the display-name sender but not asserted across the time+name sender, so a parallel production reopened the class. Make the oracle test each SEMANTIC RULE against the FULL set of productions that can match it: assert `pronoun-in-sender -> content` across every sender production (display-name AND time+name AND delimiter forms), and assert `explicit-evidence row -> kept` regardless of gate-body emptiness. When a rule is verified against the whole grammar rather than one production, a second production carrying the same shape cannot silently reopen it. Concretely: add `On <date> <time> I wrote:` to the sender cross-product with a content verdict, and make the junk fallback evidence-aware (do not classify `no_new_content` while `resolution_text` or other explicit fields are present).

effect-trace: two sender/junk productions apply the drop decision without the semantic guard the parallel production already has, so a real question (`:129`) and a real resolution row (`:738`) are removed from `source_material` -- content loss vs main, not bounded by subset-of-main. boundary-probe: the oracle covers the rule on one production per class but not across parallel productions; that gap is exactly where both regressions sit. Verdict: MAJOR -- the method is now correct; make the oracle rule-complete across productions and evidence-aware, and this converges instead of reopening.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `90d6aecb` (round 8). Both round-7 regressions are fixed and I verified them (`:129` time+name pronoun keeps the question; `:738` resolution_text row survives). But round 8 reopened the lost-question class through two MORE sender tokens, and after 8 rounds that is now a root finding, not another enumeration gap. MAJOR; `claude-review=failure`. Pure-reviewer.**

**Confirmed content-loss by probe (both drop the customer question; control passes):**
- `:132` -- `On Monday 09:15 It wrote:\nError 500. How do I fix it?` -> ROW DROPPED (`It` accepted as sender).
- `:131` -- `On Monday, The System wrote:\nError 500. How do I fix it?` -> ROW DROPPED (`The System` accepted as a two-token name).
- control `On Monday, Jane Agent wrote:\nOld body.` correctly drops the old body -- the grammar works for real senders.

**The root, stated plainly after 8 rounds:** the reply-header recognizer decides a quote by pattern-matching capitalized tokens before `wrote:`, then EXCLUDES the following content. So the guard's real question is `is this token a person/sender?` -- an open-ended NER judgment -- and every round Codex supplies another non-person token the enumeration missed: `I` -> `We` -> `It` -> `The System`. That denylist is unbounded; enumerating it in the oracle will never converge, because the failure is over-MATCHING ordinary prose as a header, and you cannot enumerate all prose that isn't a sender.

**The fix that actually closes the class -- flip to evidence-required, default-to-content.** Only treat `On <date>, <X> wrote:` (and `-----Original Message-----`) as a real quoted-reply header when there is CORROBORATING evidence it is one: an email address in the header, or the next non-blank line is a quoted body (`>`/quoted block), or `X` carries person+contact evidence. Absent that evidence, it is ordinary customer prose and its content is KEPT. Same for signatures: require terminal contact evidence, else keep. This makes the scanner fail toward under-exclusion (<= main, safe) by construction, so an ambiguous `On Monday, The System wrote:` defaults to content and the lost-question class is closed once, not token by token.

**Read the two directions honestly (they trade off):** the 3 other new threads are under-exclusion -- `:1018` single-name signature, `:983` `From:`+`Date:` original-message, `:152` non-ASCII names -- all <= main (junk leaks no worse than today, no content lost). Requiring stronger evidence to skip will INCREASE these (more real signatures/quotes leak for lack of evidence). That is the correct bias for a FAQ-mining product: a leaked signature line is cheap; a dropped customer question is the product failing. Accept more <=main under-exclusion to eliminate all over-exclusion.

**Decision, since this is round 8 and not converging:** either adopt the evidence-required / default-to-content design (a bounded refactor of the recognizers -- the real convergence), or freeze the current best-effort, document that it can still drop content on non-person sender prose, and stop precision-chasing. Continuing to add the next non-person token to the oracle is neither. I would take the evidence-required refactor. Verdict: MAJOR -- close the content-loss class by construction, do not enumerate the next sender.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `13741c23` (round 9). The denylist grew (I/We -> +determiners/pronouns), so `It`/`The System`/`Our Team` now keep content -- but this is still enumeration, and it still drops real questions. MAJOR; `claude-review=failure`. Pure-reviewer.**

**Confirmed content-loss on FIVE ordinary support-ticket senders (direct probe at head), so this is not a contrived edge:**
- `On Monday, Support Team wrote:\nError 500. How do I fix it?` -> ROW DROPPED
- `On Monday, Acme Corp wrote:` -> ROW DROPPED
- `On Monday, Billing Department wrote:` -> ROW DROPPED
- `On Monday, Customer Success wrote:` -> ROW DROPPED
- `On Monday, Escalations Queue wrote:` -> ROW DROPPED
Each is two capitalized tokens that are not a person and do not start with a denylisted determiner, so the display-name sender production matches and the customer's question is discarded. The denylist cannot cover this: team/department/company/queue names are an open set.

**The shared decision seam (the one you are hunting):** every one of these threads is the SAME admit/skip decision defaulting the wrong way. The reply-header and signature recognizers currently mean `skip unless the token is on my non-person list`. Invert the default: **skip only on positive junk-evidence, else keep.** A quoted-reply header counts as a real quote only with corroborating evidence -- an email address in the header line, or the next non-blank line is a quoted body (`>`/quoted block), or `From:`+`Sent:`/`Date:`. A signature counts only with terminal contact evidence. Absent evidence, the line is ordinary customer content and is kept. One decision, evidence-gated, defaulting to content. That closes the whole content-loss class at the seam instead of per token, and it caps the scope to a finite set of positive-evidence rules rather than an unbounded person-name model.

**Direction check for the operator (raised in chat too):** the ONLY way this slice is worse than `main` is over-exclusion -- main drops nothing. So the in-scope target is `never drop customer content; remove only proven junk`, which the evidence-gated seam achieves and which is finite/closeable. Perfect person-vs-nonperson sender classification is the out-of-scope ambition that keeps spawning threads. Verdict: MAJOR -- fix at the seam (evidence-required, default-to-content), do not extend the denylist.

File: extracted_content_pipeline/support_ticket_input_package.py [hunk 1]
```diff
@@ -845,6 +928,136 @@ def _comments_text(row: Mapping[str, Any]) -> str:
...
+    signature_blank_seen = False
+    for line_index, raw_line in enumerate(lines):
+        line = raw_line.strip()
+        if line == blank_marker:
+            if skip_mode == "signature":
+                signature_blank_seen = True
+            continue
+        if not line:
+            continue
+        if _history_line_is_reply_header(lines, line_index=line_index, line=line):
```
[the remark this answers is not in the document]
Comment (canfieldjuan) [reply extracted_content_pipeline/support_ticket_input_package.py]: Fixed at af3702061. Quote prefixes are now removed only in the shared detection probe, so quoted On-wrote and Original Message headers are recognized while standalone customer > lines remain emitted. The generated package-entrypoint oracle covers one- and two-level markers across all three scalar aliases.

File: extracted_content_pipeline/support_ticket_input_package.py [hunk 2]
```diff
@@ -97,6 +97,71 @@
...
+    r"(?:jan|feb|mar|apr|may|jun|jul|aug|sep|sept|oct|nov|dec)[a-z]*\b",
+    re.IGNORECASE,
+)
+_HISTORY_REPLY_EMAIL_RE = re.compile(r"<[^>]+@[^>]+>|\b[\w.+-]+@[\w.-]+\.[a-z]{2,}\b", re.I)
+_HISTORY_PERSON_TOKEN = r"[^\W\d_][^\W\d_.'\u2019-]*"
+_HISTORY_REPLY_TIME_SENDER_RE = re.compile(
+    r"\b\d{1,2}:\d{2}\b(?:\s*[AaPp][Mm])?[\s,]+"
+    r"[A-Z][A-Za-z.'-]*(?:\s+[A-Z][A-Za-z.'-]*){1,3}\s*$")
+_HISTORY_REPLY_NAME_SENDER_RE = re.compile(
+    r",\s*[A-Z][A-Za-z.'-]*(?:\s+[A-Z][A-Za-z.'-]*){1,3}\s*$")
```
[the remark this answers is not in the document]
Comment (canfieldjuan) [reply extracted_content_pipeline/support_ticket_input_package.py]: Reconciled at af3702061 as the explicit bounded waiver recorded in the PR body. Name-only sender inference is the open-ended classifier that caused the repeated customer-content-loss class, so an uncorroborated Unicode name remains content by the accepted #2044 safety bias. Unicode reply headers are excluded when email or quote-marker evidence corroborates them, and those cases are generated across every scalar alias.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `af370206` -- the evidence-gated replacement landed and it CONVERGED. The content-loss class is closed by construction, I verified it against a sender the design never enumerated, and every remaining finding is now bounded under-exclusion (<= main). Lifting the content-loss block: `claude-review=success`. The merge now waits only on reconciling three <=main leak threads. Pure-reviewer.**

**The loop is broken -- verified both directions at head:**
- Content-loss class CLOSED: `On Monday, <X> wrote:` with no email/quote evidence keeps the customer question for every `X` I probed -- Support Team, Acme Corp, Billing Department, Customer Success, Escalations Queue, The System, Nobody Here, `I`, and crucially `Jose Garcia Team` (Unicode, never enumerated). No sender vocabulary remains, so there is no next permutation.
- Real quotes WITH evidence still excluded: email-in-header, `>` quoted block, and `From:`+`Sent:` original-message all drop the old body.
- Second side holds (my guardrail): ordinary customer prose that merely mentions an email (`Email me at jane@customer.com ...`) is KEPT -- the email evidence is bound to the reply-header shape, not global.
- The oracle is now a true independent model: verdict = `bool(header_marker or body_marker or has_email)`, i.e. keyed on evidence presence, not sender strings. That is the fix to the fixture-matrix problem.
- `unit-gate` green, extracted-content maturity ratchet green, S6C oracle test passes.

**The 3 remaining threads are all the ACCEPTED direction (under-exclusion, <= main -- old text leaks, no content lost):**
- `:925` -- HTML `<br>` extraction strips the sender email before the evidence check, so a real quote loses its evidence and leaks. Ordering bug in evidence extraction.
- `:978` -- the original-message lookahead misses `From:`->`To:`/`Cc:`/`Bcc:`->`Date:` blocks, so that block leaks.
- `:143` -- a bare blank inside a skipped run clears `skip_mode`, publishing trailing old content.
All three leak more old text; none drops customer content. They are within the tradeoff this design deliberately accepts.

**Recommendation:** `:925`, `:978`, `:143` are STRUCTURAL (evidence-extraction ordering, header-block scan, state-machine resume) -- fixable without touching vocabulary, so fixing them tightens exclusion with zero loop risk and I would take them. But they are not blockers: each is <= main. Reconcile the three (fix the cheap structural ones, and waive the Unicode name-only thread as bounded <= main with a one-line plan note, per the approved design) and `live-reconciliation` greens -> merge. From my side the blocker is gone. This is what the seam fix promised: after nine rounds of over-exclusion, the residual is bounded harmless leakage. Nice landing.

File: extracted_content_pipeline/support_ticket_input_package.py [hunk 3]
```diff
@@ -845,6 +911,127 @@ def _comments_text(row: Mapping[str, Any]) -> str:
...
+
+    raw = str(value or "").replace("\r\n", "\n").replace("\r", "\n")
+    if not raw.strip():
+        return ""
+    blank_marker = "ATLAS_HISTORY_BLANK_BOUNDARY"
+    while blank_marker in raw:
+        blank_marker += "_"
+    marked = _HISTORY_HTML_BLANK_RE.sub(f"\n{blank_marker}\n", raw)
+    marked = _HISTORY_BLANK_LINE_RE.sub(f"\n{blank_marker}\n", marked)
+    lines = support_ticket_plain_text_lines(marked).splitlines()
```
[the remark this answers is not in the document]
Comment (canfieldjuan) [reply extracted_content_pipeline/support_ticket_input_package.py:931]: Fixed at e3c17582f. Scalar history now entity-protects angle-bracket email addresses before line-preserving HTML extraction, so the parser retains sender evidence instead of treating it as a tag. The package-entrypoint oracle generates plain, <br>, and <br/> separators across all aliases, quote-marker depths, and email-evidence states.

File: extracted_content_pipeline/support_ticket_input_package.py [hunk 4]
```diff
@@ -845,6 +911,127 @@ def _comments_text(row: Mapping[str, Any]) -> str:
...
+    return match.group("history_quoted_line").strip(), True
+
+
+def _history_line_is_reply_header(
+    lines: Sequence[str], *, line_index: int, line: str, blank_marker: str
+) -> bool:
+    probe, header_is_quoted = _history_line_probe(line)
+    if _HISTORY_ORIGINAL_MESSAGE_RE.fullmatch(probe):
+        tail: list[str] = []
+        for item in lines[line_index + 1 : line_index + 5]:
```
[the remark this answers is not in the document]
Comment (canfieldjuan) [reply extracted_content_pipeline/support_ticket_input_package.py]: Fixed at e3c17582f. Original Message recognition now scans a bounded 12-line run of arbitrary RFC-shaped header names, stopping on blank/non-header text and requiring From plus Sent/Date. The oracle covers zero and five intermediate fields, Sent/Date, quote markers, and every scalar alias.

File: extracted_content_pipeline/support_ticket_input_package.py [hunk 5]
```diff
@@ -97,6 +97,54 @@
...
+    r"^(?:thanks|thank you|best|regards|sincerely|cheers)[,!]?$", re.I
+)
+_HISTORY_SIGNATURE_LOOKAHEAD = 10
+_HISTORY_SIGNATURE_CONTACT_RE = re.compile(
+    r"[\w.+-]+@"
+    r"[\w.-]+\.[a-z]{2,}|https?://|www\.|(?:^|\s)\+?\d[\d .()/-]{6,}\d(?:\s|$)",
+    re.IGNORECASE,
+)
+_HISTORY_EVENT_POLICY = {
+    "blank": ("", False),
```
[the remark this answers is not in the document]
Comment (canfieldjuan) [reply extracted_content_pipeline/support_ticket_input_package.py:146]: Waived with operator approval and recorded in the plan/PR body. Accepted issue #2044 explicitly requires quote/signature exclusion only until blank or a likely new-message boundary, so removing blank resumption would change the accepted contract. The generated oracle continues to prove plain, Unicode, and HTML blank resumption in both skip modes.

**Review by canfieldjuan — COMMENTED** [review]
**Re-review at `e3c17582` -- converged. The content-loss class stays closed by construction, the two structural leaks are fixed, and the one remaining review request is a documented, justified waiver, not a silent gap. LGTM; `claude-review=success`. Pure-reviewer, so the merge is the owner's -- it is green and clean.**

Verified at head:
- Content-loss class still closed: `On Monday, <X> wrote:` with no evidence keeps the customer question for Support Team, The System, Nobody Here, and the novel Unicode `Jose Garcia Team`.
- `:925` fixed: an HTML `<br>` reply header with an angle-bracket email now excludes the old body (the email evidence is protected before HTML extraction).
- `:978` fixed: original-message recognition scans a bounded header-field run, accepting `Sent:`/`Date:` after optional `To:`/`Cc:`/`Bcc:`.
- `:143` waived correctly and DOCUMENTED: bare-blank resumption is retained intentionally because accepted issue #2044 names blank as a valid end to a quote/signature run (plan line 44). This is a request to change the accepted product contract, so waiving it -- reconciled, not silently dropped -- is the right call. The residual is under-exclusion (<= main), never content loss.
- The safety bias is stated in the plan (line 11): ambiguous free text is customer content by default; more old-text leakage acceptable, customer-content loss not. The oracle is now event/evidence-keyed (grammar-derived cross-product), not sender vocabulary.
- unit-gate green, extracted-content maturity ratchet green, live-reconciliation green, 0 unresolved threads, status CLEAN.

**Arc note:** this is the payoff after eleven rounds. Enumeration (denylist then allowlist over sender vocabulary) never converged and kept dropping customer questions; the evidence-gated event/state redesign closed the content-loss class in one pass and reduced the residual to bounded <=main leakage plus one documented contract waiver. That is exactly the open-category -> evidence-gate + asymmetric-safe-default pattern now codified in `docs/GUARD_CLASS_CLOSURE.md` / `AGENTS.md` 3k.2 (#2077). Clean landing. LGTM.
